src/main/java/io/xcherry/core/client/Client.java

```diff
@@ -334,4 +337,32 @@ private void publishToLocalQueue(final String processId, final LocalQueueMessage
 
         basicClient.publishToLocalQueue(request);
     }
+
+    /**
+     * Wait for a process execution to complete within the specified timeout seconds.
+     *  1. If the process execution is still running after the timeout seconds, the response will be returned with the timeout field being true.
+     *  2. If the process execution has stopped by system (e.g., due to re-balancing after increasing/decreasing the shards in the server), the response will be returned with the stopBySystem field being true.
+     *  3. In other cases, the current status of the process execution will be returned.
+     *
+     * @param processId         a unique identifier used to differentiate between different executions of the same process type.
+     */
+    public ProcessExecutionWaitForCompletionResponse waitForProcessCompletion(final String processId) {
```

Comment:
How about just returning status and throw exception?
* WaitTimeoutException extends ClientSideException: for the process is still running
* ServerSideException for the reblancing and others

Reply:
Why WaitTimeoutException should extend ClientSideException? It it extends the ClientSideException, we need to build FeignClientException to pass into, which is cumbersome. I think just extending RuntimeException should be fine here.